habitat/tasks/eqa/eqa_task.py

```diff
@@ -48,77 +46,225 @@ class EQAEpisode(NavigationEpisode):
     )
 
 
+@registry.register_sensor
 class QuestionSensor(Sensor):
-    def __init__(self, **kwargs):
+    def __init__(self, dataset, **kwargs):
         self.uuid = "question"
         self.sensor_type = SensorTypes.TEXT
-        # TODO (maksymets) extend gym observation space for text and metadata
-        self.observation_space = spaces.Discrete(0)
+        self._dataset = dataset
+
+        self.observation_space = spaces.Discrete(
+            len(dataset.get_questions_vocabulary())
+        )
 
-    def _get_observation(
+    def get_observation(
         self,
         observations: Dict[str, Observations],
         episode: EQAEpisode,
-        **kwargs
+        **kwargs,
     ):
-        return episode.question.question_text
+        return self._dataset.get_questions_vocabulary()[
+            episode.question.question_text
+        ]
+
+
+@registry.register_measure
+class CorrectAnswer(Measure):
```

Comment:
This should be a sensor?  It's simpler for learning if it's in the sensor land.

Reply:
@erikwijmans, we expect to have in sensors only data that will be input for a model, as it's discoverable through observation space. This way we can avoid leakage by design. For PointGoal, we use distance to target as reward (supervision) and that wasn't sensor.